### PR TITLE
Cleaning before state page

### PR DESCRIPTION
**Pull request by octopusinvitro**
This is a previous cleaning step before dealing with showing all people in child areas on a State page.

state-page

### CODE REVIEW

File: views/place.erb [hunk 1]
```diff
@@ -30,7 +30,7 @@
           <h1 class="person__name"><%= @page.title %></h1>
           <h2>
             <%= @page.place.type_name %>
-            <%= " #{(@page.legislature_name)}" if @page.place.is_child_area? %>
+            <%= " (#{@page.legislature_name})" if @page.place.is_child_area? %>
```
Comment (mhl) [inline views/place.erb]: This fix is unrelated to the rest of the commit, I think, so it should be in a separate commit.

File: lib/membership_csv/people.rb [hunk 2]
```diff
@@ -22,6 +22,10 @@ def none?(id)
       find_single(id).nil?
     end
 
+    def find_all_by_mapit_area(mapit_id)
+      find_all.select { |person| person.area.id == mapit_id if person.area }
```
Comment (mhl) [inline lib/membership_csv/people.rb]: We've talked about this pattern quite a lot - how could you improve this implementation?

**Comment by octopusinvitro** [comment]
@mhl Feedback addressed :sake: :violin:

File: lib/membership_csv/people.rb [hunk 3]
```diff
@@ -42,6 +46,17 @@ def id_to_person
       @id_to_person ||= all_people.map { |person| [person.id, person] }.to_h
     end
 
+    def mapit_id_to_person
+      @mapit_id_to_person ||= all_people.each_with_object({}) { |person, memo| update(person, memo) }
+    end
+
+    def update(person, memo)
```
Comment (mhl) [inline lib/membership_csv/people.rb]: I think this method name is too generic - how about something like `update_people_for_area`?

File: lib/membership_csv/people.rb [hunk 4]
```diff
@@ -42,6 +46,17 @@ def id_to_person
...
+      return unless person.area
+      mapit_id = person.area.id.to_s
+      memo[mapit_id] = [] unless memo[mapit_id]
+      memo[mapit_id] << person
```
Comment (mhl) [inline lib/membership_csv/people.rb]: I think these three lines could be reduced to:

```
(memo[person.area.id.to_s] ||= []) << person
```

(If you don't want to go that far, at least using `memo[mapit_id] ||= []` would be a good change.)
